include/ensmallen_bits/function/static_checks.hpp

```diff
@@ -375,7 +375,7 @@ inline void CheckSparseFunctionTypeAPI()
 /**
  * Perform checks for the ArbitraryFunctionType API.
  */
-template<typename FunctionType, typename MatType>
+template<typename FunctionType, typename MatType, std::size_t I = 0U>
```

Comment:
Do you mind to use `0` instead of `0U` here and  you used that in a couple of other places as well, I know this is a preference, but ideally I like to keep the same style across the entire codebase.